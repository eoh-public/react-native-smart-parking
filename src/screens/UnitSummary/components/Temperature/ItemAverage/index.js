import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';

import { Colors } from '../../../../../configs';
import Text from '../../../../../commons/Text';

const ItemAverage = memo((props) => {
  const { svgMain, value } = props;
  return (
    <View style={styles.container}>
      {svgMain}
      <Text size={14} color={Colors.Gray9} style={styles.title}>
        {value}
      </Text>
    </View>
  );
});

export default ItemAverage;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 9,
    paddingLeft: 8,
    paddingRight: 16,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    flexDirection: 'row',
    marginRight: 8,
    alignItems: 'center',
  },
  title: {
    marginLeft: 12,
  },
});
