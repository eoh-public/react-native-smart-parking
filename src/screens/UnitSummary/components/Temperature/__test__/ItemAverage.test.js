import React from 'react';
import { act, create } from 'react-test-renderer';

import ItemAverage from '../ItemAverage';

describe('Test ItemAverage', () => {
  let tree;
  test('render ItemAverage', async () => {
    act(() => {
      tree = create(<ItemAverage />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
