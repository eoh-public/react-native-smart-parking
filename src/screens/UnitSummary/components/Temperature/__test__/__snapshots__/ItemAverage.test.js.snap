// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemAverage render ItemAverage 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E8E8E8",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginRight": 8,
      "paddingLeft": 8,
      "paddingRight": 16,
      "paddingVertical": 9,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 14,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 12,
        },
      ]
    }
  />
</View>
`;
